admin_guide/managing_networking.adoc

```diff
@@ -378,6 +418,227 @@ If only some of the nodes in your cluster are capable of claiming the specified
+different specification:
+
+.Example Pod Definition for an Egress Router HTTP Proxy
+====
+----
+apiVersion: v1
+kind: Pod
+metadata:
+  name: egress-http-proxy
+  labels:
+    name: egress-http-proxy
+  annotations:
+    pod.network.openshift.io/assign-macvlan: "true" <1>
+spec:
+  initContainers:
+  - name: egress-router-setup
+ifdef::openshift-enterprise[]
+    image: registry.access.redhat.com/openshift3/ose-egress-router
+endif::openshift-enterprise[]
+ifdef::openshift-origin[]
+    image: openshift/origin-egress-router
+endif::openshift-origin[]
+    securityContext:
+      privileged: true
+    env:
+    - name: EGRESS_SOURCE <2>
+      value: 192.168.12.99
+    - name: EGRESS_GATEWAY <3>
+      value: 192.168.12.1
+    - name: EGRESS_ROUTER_MODE <4>
+      value: http-proxy
+  containers:
+  - name: egress-router-proxy
+ifdef::openshift-enterprise[]
+    image: registry.access.redhat.com/openshift3/ose-egress-router-http-proxy
+endif::openshift-enterprise[]
+ifdef::openshift-origin[]
+    image: openshift/origin-egress-router-http-proxy
+endif::openshift-origin[]
+    env:
+    - name: EGRESS_HTTP_PROXY_DESTINATION <5>
+      value: |
+        !*.example.com
+        !192.168.1.0/24
+        *
+----
+<1> The `pod.network.openshift.io/assign-macvlan annotation` creates a Macvlan
+network interface on the primary network interface, and then moves it into the
+pod's network name space before starting the *egress-router* container. Preserve
+the the quotation marks around `"true"`. Omitting them will result in errors.
+<2> An IP address from the physical network that the node itself is on and is
+reserved by the cluster administrator for use by this pod.
+<3> Same value as the default gateway used by the node itself.
+<4> This tells the egress router image that it is being deployed as
+part of an HTTP proxy, and so it should not set up iptables
+redirecting rules.
+<5> A string or YAML multi-line string specifying how to configure the
+proxy. Note that this is specified as an environment variable in the
+HTTP proxy container, not with the other environment variables in the
+init container.
+====
+
+You can specify a variety of things for
+`EGRESS_HTTP_PROXY_DESTINATION`. The simplest is to just specify `*`,
+meaning "allow connections to all remote destinations". Other than
+that, each line specifies one group of connections to allow or deny:
+
+- An IP address (eg, `192.168.1.1`) will allow connections to that IP address.
+- A CIDR range (eg, `192.168.1.0/24`) will allow connections to that CIDR range.
+- A hostname (eg, `www.example.com`) will allow proxying to that host.
+- A domain name preceded by `\*.` (eg, `*.example.com`) will allow proxying to that domain and all of its subdomains.
+- A `!` followed by any of the above will deny connections rather than allowing them
+- If the last line is `*`, then anything that hasn't been denied will be allowed. Otherwise, anything that hasn't been allowed will be denied.
```

Comment:
When you are using a forward HTTP proxy, the client has to be aware of that and talk to the proxy differently.  https://en.wikipedia.org/wiki/HTTP_tunnel#HTTP_CONNECT_tunneling

Basically, the client asks the proxy to open a connection over HTTP and then the client sends the HTTPS traffic through a tunneled connection.